### ISSUE TITLE
MiniImagenet results

### ISSUE THREAD

**Issue by xu-ji** [optimass/Maximally_Interfered_Retrieval#6]
Hi, did you run any experiments on MiniImagenet for either doing nothing (finetuning) or stationary data?
I ask because I am getting 8.4% accuracy for 1 epoch of stationary training (rises to 33.1% after 5 epochs), which is much lower than 15.4% for ER and 16.8% for MIR (1 epoch, 1 iteration, 10k total buffer size, running your code). I am wondering how this can be? And whether you observed anything similar.

**Comment by eugenium** [comment]
Hi Xu, thanks for your interest. We haven't looked at other basleines for minimagenet, but I just ran the iid baseline and indeed I get around 10.3%. Here is the script I used https://gist.github.com/eugenium/8b902e28ea5082f7f7e39d7ae2c32376

It is indeed suprising. My guess would be the discrepancy is likely due to subleties of the evaluation protocol of aggregating all tasks accuracy. For example in non-iid training with ER some earlier tasks may obtain very high performance thus drowning out poor performance in later tasks. This phenomenon is not really present in cases with short task sequences/few classes (like mnist split, cifar split) since poor performance on later tasks would heavily decrease overall accuracy. We are actually currently looking into some additional metrics to use in long tasks sequences that can potentially take these imbalanced issues into account.

**Comment by xu-ji** [comment]
It seems what you're implying is there is a replay distribution that is even better than iid for MiniImageNet. That the buffer is not perfectly balanced is good, because it means (a subset) of classes receive disproportionately large additional training effort.

A good check for this could be to use a buffer with equal budget per class. Then, if ER/MIR still performs much better than iid, that's strange. Since replay can not use classes not yet seen in training, it should really be worse than iid.

Thank you

**Comment by pclucas14** [comment]
I think it's worth noting that doing ER will allow the model to 'train on more data' since the batch size essentially doubles (assuming the buffer batch size == incoming batch size). A good baseline would be to run iid training for 2 epochs with 2x bigger batch size and see what happens. 

-Lucas

**Comment by xu-ji** [comment]
I agree total number of training steps is definitely a material factor. With ER you take 2x the steps (though it could also be 1x with different implementation - another good baseline). Regarding absolute number of samples/batch size, I'm not so sure, because training signal is averaged across samples, so you would expect the data distribution to be the material factor. And ER would suffer compared to iid in this respect due to absence of unseen classes.
